engine/src/main/java/org/terasology/engine/subsystem/lwjgl/LwjglResolution.java

```diff
@@ -15,42 +15,105 @@
  */
 package org.terasology.engine.subsystem.lwjgl;
 
-import org.lwjgl.opengl.DisplayMode;
+import org.lwjgl.glfw.GLFWVidMode;
 import org.terasology.engine.subsystem.Resolution;
 
+import java.util.Objects;
+
 public final class LwjglResolution implements Resolution {
 
-    private final DisplayMode displayMode;
+    private int width;
+    private int height;
+    private int redBits;
+    private int greenBits;
+    private int blueBits;
+    private int refreshRate;
+
+    public LwjglResolution(GLFWVidMode vidMode) {
+        this.width = vidMode.width();
+        this.height = vidMode.height();
+        this.redBits = vidMode.redBits();
+        this.greenBits = vidMode.greenBits();
+        this.blueBits = vidMode.blueBits();
+        this.refreshRate = vidMode.refreshRate();
+    }
+
+    public LwjglResolution(int width, int height, int redBits, int greenBits, int blueBits, int refreshRate) {
+        this.width = width;
+        this.height = height;
+        this.redBits = redBits;
+        this.greenBits = greenBits;
+        this.blueBits = blueBits;
+        this.refreshRate = refreshRate;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public int getRedBits() {
+        return redBits;
+    }
 
-    public LwjglResolution(DisplayMode displayMode) {
-        this.displayMode = displayMode;
+    public int getGreenBits() {
+        return greenBits;
     }
 
-    public DisplayMode getDisplayMode() {
-        return displayMode;
+    public int getBlueBits() {
+        return blueBits;
+    }
+
+    public int getRefreshRate() {
+        return refreshRate;
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (o == null) {
+            return false;
+        } else if (o instanceof GLFWVidMode){
```

Comment:
Very trivial, but a missing space here between `){` - haven't looked too closely for such code style issues elsewhere, but it may be worth doing a little Checkstyle pass per changed class in your IDE